<?xml version="1.0"?>
<robot xmlns:xacro="http://www.ros.org/wiki/xacro">

  <xacro:macro name="kwr75_force_sensor_ros2_control" params="name prefix com_port:=/dev/ttyUSB0 use_mock_hardware:=false mock_sensor_commands:=false">
    <xacro:unless value="${use_mock_hardware}">
      <ros2_control name="${name}" type="sensor">
        <hardware>
          <!-- <xacro:if value="${use_mock_hardware}">
            <plugin>mock_components/GenericSystem</plugin>
            <param name="mock_sensor_commands">${mock_sensor_commands}</param>
            <param name="position_state_following_offset">0.0</param>
          </xacro:if> -->
          
          <plugin>kwr75_force_sensor/KWR75ForceSensorHardwareInterface</plugin>
          <param name="com_port">${com_port}</param>
          <param name="baud_rate">460800</param>
          
        </hardware>

        <sensor name="${prefix}kwr75_force_sensor">
          <state_interface name="force.x"/>
          <state_interface name="force.y"/>
          <state_interface name="force.z"/>
          <state_interface name="torque.x"/>
          <state_interface name="torque.y"/>
          <state_interface name="torque.z"/>
          <param name="frame_id">${prefix}fts_link</param>
          <param name="fx_range">200.0</param>
          <param name="fy_range">200.0</param>
          <param name="fz_range">200.0</param>
          <param name="tx_range">8.0</param>
          <param name="ty_range">8.0</param>
          <param name="tz_range">8.0</param>
        </sensor>
      </ros2_control>
    </xacro:unless>
  </xacro:macro>

</robot>